scylla/tests/ccm_integration/keep_alive.rs

```diff
@@ -0,0 +1,276 @@
+use crate::common::utils::test_with_3_node_cluster;
+use scylla::client::session::Session;
+use scylla::client::session_builder::SessionBuilder;
+
+use scylla_proxy::{
+    Condition, ProxyError, Reaction, RequestOpcode, RequestReaction, RequestRule, ShardAwareness,
+    WorkerError,
+};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc::{self, UnboundedReceiver};
+use tokio::time::sleep;
+
+/// Waits for a specified number of keep-alive messages to be received within a given timeout period.
+///
+/// # Arguments
+///
+/// * `request_rx` - An `UnboundedReceiver` that receives the keep-alive messages.
+/// * `expected_number_of_keep_alive` - The number of keep-alive messages expected to be received.
+/// * `timeout` - The maximum duration to wait for the expected number of keep-alive messages.
+///
+/// # Panics
+///
+/// This function will panic if the number of received keep-alive messages does not match the
+/// expected number within the timeout period.
+///
+async fn wait_for_keep_alive<T>(
+    mut request_rx: UnboundedReceiver<T>,
+    expected_number_of_keep_alive: u32,
+    timeout: Duration,
+) {
+    let mut total_keep_alives: u32 = 0;
+    let start = tokio::time::Instant::now();
+    while total_keep_alives < expected_number_of_keep_alive && start.elapsed() < timeout {
```

Comment:
I'm not sure this is the best idiomatic  way of doing such a thing

Reply:
It is not. I'd do it roughly like that (didn't test the code, so it may not work):
```rust
tokio::time::timeout(timeout,  async move {
    for _ in 0..expected_number_of_keep_alive {
        request_rx.recv().await.unwrap();
    }
});
```